addons/website/static/src/snippets/s_popup/000.js

```diff
@@ -397,7 +405,98 @@ publicWidget.registry.cookies_bar = PopupWidget.extend({
             }
         }
         setUtmsHtmlDataset();
-    }
+    },
+    /**
+     * Reopens the cookies bar if it was closed.
+     *
+     * @private
+     */
+    _onShowCookiesBar() {
+        const modalEl = this.el.querySelector(".modal");
+        const currCookie = cookie.get(this.el.id);
+
+        if (currCookie && JSON.parse(currCookie).optional || !this._popupAlreadyShown) {
+            return;
+        }
+        $(modalEl).modal("show");
+
+        // The cookies bar remains hidden, most probably because of the browser
+        // or an extension: notify the user because "nothing happens when I
+        // click" is never good.
+        if (!isVisible(modalEl)) {
+            window.alert(_t("Our cookies bar was blocked by your browser or an extension."));
+            return;
+        }
+        modalEl.focus();
+    },
+});
+
+publicWidget.registry.cookiesApproval = publicWidget.Widget.extend({
+    selector: "[data-need-cookies-approval]",
+    events: {
+        "add_cookies_warning": "_onAddCookiesWarning",
+    },
```

Comment:
There is something strange about this, if this is right it needs comments:
- That `add_cookies_warning` event is triggered... only in the code that adds `data-need-cookies-approval` on the element too. So this public widget could not have been started to listen to that event since he just receive the target selector. Or setting `data-need-cookies-approval` in `manageIframeSrc` is just useless?

Reply:
The event is not needed on elements that were already marked when the widget is started (in which case the warning is added on `willStart()`), but both kinds of elements (premarked & marked in `manageIframeSrc`) set the same warning and need the same processing, so I do them both here.

> Or setting data-need-cookies-approval in manageIframeSrc is just useless?

It mostly is useless, but not totally: videos created with `_generateIframe()` set their src with `manageIframeSrc`, and the video size container is hidden if `data-need-cookies-approval` is set.

Reply:
> but both kinds of elements (premarked & marked in manageIframeSrc)

That was my point: I don't understand how the non-premarked case works:
1. You call `manageIframeSrc`
2. It adds `data-need-cookies-approval`
3. It triggers the `add_cookies_warning`... which does nothing since there were public widget created for this element during page loading since it did not have the `data-need-cookies-approval` yet.

... but I guess it works because you expect to catch the event on the container which has `data-need-cookies-approval` and not on the iframe itself? I think it needs comments somewhere if that is the case. Also `_onAddCookiesWarning`: should not the event propagation be stopped?

> in which case the warning is added on willStart()

Actually do that in the `start`. There is no difference in practice but `willStart` in publicWidget is probably always wrong from a consistency point of view. For normal widget, the point was:
- willStart: before the widget html is rendered and added in the DOM, do some async stuff which does not impact the DOM (typically fetching data needed for the rendering).
- start: after the widget is added in the DOM, do stuff.

Since for public widget, we attach ourselves on existing DOM, willStart rarely make sense. In this case, even in the widget case, you want to use `start` anyway: you are doing something to the DOM.